File: 1108.defanging-an-ip-address.cpp

#include <iostream>
#include <algorithm>
#include <regex>
using namespace std;
/*
 * @lc app=leetcode id=1108 lang=cpp
 *
 * [1108] Defanging an IP Address
 */

// @lc code=start
class Solution
{
public:
    string defangIPaddr(string address)
    {
        return regex_replace(address, regex("[.]"), "[.]");
    }
};
// @lc code=end
int main()
{
    Solution ob;

    string address = "1.1.1.1";

    string answer = ob.defangIPaddr(address);
    cout << answer << endl;
}

File: 1108.defanging-an-ip-address.js
/*
 * @lc app=leetcode id=1108 lang=javascript
 *
 * [1108] Defanging an IP Address
 */

// @lc code=start
/**
 * @param {string} address
 * @return {string}
 */
var defangIPaddr = function (address) {

    function escapeRegExp(string) {
        return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
    }
    function replaceAll(str, match, replacement) {
        return str.replace(new RegExp(escapeRegExp(match), 'g'), () => replacement);
    }

    return replaceAll(address, ".", "[.]");
};
// @lc code=end

